packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -192,9 +192,13 @@ export class SelectedNetworkController extends BaseController<
     this.#setNetworkClientIdForDomain(domain, networkClientId);
   }
 
-  getNetworkClientIdForDomain(domain: Domain) {
+  getNetworkClientIdForDomain(domain: Domain): NetworkClientId {
     if (this.state.perDomainNetwork) {
-      return this.state.domains[domain];
+      const id = this.state.domains[domain];
+      if (id === undefined) {
+        return this.state.domains[METAMASK_DOMAIN];
+      }
+      return id;
```

Comment:
This is even shorter and would avoid using `this.state.domains[METAMASK_DOMAIN]` if `this.state.domains[domain]` is an empty string as @legobeat pointed out above.

``` suggestion
      return this.state.domains[domain] ?? this.state.domains[METAMASK_DOMAIN];